API/services/userService.js

```diff
@@ -0,0 +1,36 @@
+const userDao = require('../models/userDao');
+const { BaseError } = require('../utils/error');
+const axios = require('axios');
+const jwt = require('jsonwebtoken');
+
+const signInKakao = async (kakaoToken) => {
+  const result = await axios.post('https://kapi.kakao.com/v2/user/me', null, {
+    headers: {
+      'Content-type': 'application/x-www-form-urlencoded;charset=utf-8',
+      Authorization: `Bearer ${kakaoToken}`,
+    },
+  });
+  const { data } = result;
+  const nickname = data.properties.nickname;
+  const email = data.kakao_account.email;
+  const kakaoId = data.id;
+  let user = await userDao.getUserByKakaoId(kakaoId);
+
+  if (!user) {
+    await userDao.createUser(kakaoId, nickname, email);
+    user = await userDao.getUserByKakaoId(kakaoId);
+  }
```

Comment:
- 적절하게 공백을 줘서 가독성을 올려주세요.
- user가 존재하지 않을 때 user를 생성하고, 다시 user 정보를 가져오는데, 그러지 말고 createUser의 결과가 user의 정보가 되게끔 수정해 주세요.
```suggestion
  });
  
  const { data } = result;
  const nickname = data.properties.nickname;
  const email = data.kakao_account.email;
  const kakaoId = data.id;
  
  let user = await userDao.getUserByKakaoId(kakaoId);

  if (!user) {
    user = await userDao.createUser(kakaoId, nickname, email);
  }
```